src/components/Checkbox/Checkbox.tsx

```diff
@@ -43,125 +44,138 @@ export interface BaseProps {
   onBlur?(): void;
 }
 
+export interface CheckboxHandles {
+  focus(): void;
+}
+
 export interface CheckboxProps extends BaseProps {}
-type CombinedProps = CheckboxProps & WithAppProviderProps;
 
 const getUniqueID = createUniqueIDFactory('Checkbox');
 
-class Checkbox extends React.PureComponent<CombinedProps, never> {
-  private inputNode = React.createRef<HTMLInputElement>();
-
-  handleInput = () => {
-    const {onChange, id, disabled} = this.props;
-
-    if (onChange == null || this.inputNode.current == null || disabled) {
+const ImperitaveCheckbox: RefForwardingComponent<
```

Comment:
I think we'd be able to sidestep this explicit typing stuff if we define this inline rather than a separate function

```
export const Checkbox = React.forwardRef(function Checkbox({
  /* Props here */
}) {
  /* Blah */
});

Reply:
I couldn't get this to work for some reason. I'll give it another go.